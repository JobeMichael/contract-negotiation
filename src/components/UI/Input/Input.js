import React from "react";
import * as Styled from "./Input.styles";

const Input = ({ label, name, errorMsg, ...rest }) => {
  return (
    <Styled.Filed>
      <label>{label}</label>
      <div>
        <input
          name={name}
          autoComplete="off"
          className="input-element"
          {...rest}
        />
        {errorMsg && <p>{errorMsg}</p>}
      </div>
    </Styled.Filed>
  );
};

export default Input;
