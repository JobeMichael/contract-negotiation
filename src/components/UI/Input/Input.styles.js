import styled from "styled-components";

export const Filed = styled.div`
  display: flex;
  align-items: center;
  position: relative;
  flex-direction: column;

  label {
    font-size: 14px;
    margin-right: 8px;
    align-self: baseline;
    text-transform: uppercase;
    font-weight: 700;
  }

  input {
    outline: none;
    border: 2px solid #29bdba;
    background-color: transparent;
    font-size: 20px;
    padding: 6px 10px;
    display: block;
    width: 250px;
    box-sizing: border-box;
    font-weight: 700;
    margin-top: 12px;
    line-height: 1.5;
  }

  input:focus {
    outline: none;
  }

  input.invalid {
    border: 2px solid red;
    background-color: #fda49a;
  }

  input::-webkit-outer-spin-button,
  input::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  p {
    position: absolute;
    margin: 0;
    font-size: 12px;
    margin-top: 6px;
    font-weight: 600;
    color: red;
  }

  p::before {
    display: inline;
    content: "âš  ";
  }
`;
