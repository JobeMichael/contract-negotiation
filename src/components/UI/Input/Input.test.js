import { render } from "@testing-library/react";
import React from "react";
import Input from "./Input";

const testProps = {
  label: "Amount",
  name: "amount",
};

describe("<Input/>", () => {
  it("renders Input correctly", () => {
    const { asFragment } = render(<Input {...testProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Input with error", () => {
    const { asFragment } = render(
      <Input {...{ ...testProps, errorMsg: "error" }} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
