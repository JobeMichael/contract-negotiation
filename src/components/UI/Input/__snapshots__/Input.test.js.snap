// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> renders Input correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm dTemXD"
  >
    <label>
      Amount
    </label>
    <div>
      <input
        autocomplete="off"
        class="input-element"
        name="amount"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Input/> renders Input with error 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm dTemXD"
  >
    <label>
      Amount
    </label>
    <div>
      <input
        autocomplete="off"
        class="input-element"
        name="amount"
      />
      <p>
        error
      </p>
    </div>
  </div>
</DocumentFragment>
`;
